core/src/main/scala/org/http4s/Scheme.scala

```diff
@@ -0,0 +1,59 @@
+package org.http4s
+
+import cats.{Order, Show}
+ import org.http4s.internal.parboiled2.CharPredicate.{Alpha, Digit}
+import org.http4s.internal.parboiled2.{Parser => PbParser}
+import org.http4s.parser.Http4sParser
+import org.http4s.util.{hashLower, Writer}
+
+final class Scheme private (val value: String) extends Comparable[Scheme] {
```

Comment:
Intentionally does not override `Renderable`, because I want to encourage type classes over inheritance.

Intentionally not a case class, because the value it could extract does not have similar equality semantics to the type itself.